pinot-core/src/main/java/org/apache/pinot/core/plan/maker/InstancePlanMakerImplV2.java

```diff
@@ -158,7 +170,8 @@ public PlanNode makeSegmentPlanNode(IndexSegment indexSegment, QueryContext quer
     } else {
       indexSegment.prefetch(queryContext.getColumns());
     }
-    if (QueryContextUtils.isAggregationQuery(queryContext)) {
+    if (QueryContextUtils.isAggregationQuery(queryContext) ||
```

Comment:
Unclear on why the distinct query check needed here? There is also an `else` part.